bakery/mgostorage/rootkey.go

```diff
@@ -226,7 +226,23 @@ func (s *rootKeyStorage) Get(id string) ([]byte, error) {
 	return key.RootKey, nil
 }
 
-func (s *rootKeyStorage) getFromMongo(id string) (rootKey, error) {
+func (s *rootKeyStorage) getFromMongo(id []byte) (rootKey, error) {
+	var key rootKey
+	err := mgoCollectionFindId(s.coll, id).One(&key)
+	if err != nil {
+		if err == mgo.ErrNotFound {
+			return s.getLegacyFromMongo(string(id))
+		}
+		return rootKey{}, errgo.Notef(err, "cannot get key from database")
+	}
+	// TODO migrate the key from the old format to the new format.
```

Comment:
Yeah, it *might* be OK - I'm sure I thought of a reason earlier why we should do it.
I'll leave the TODO in for the moment.